subprojects/docs/src/docs/userguide/tutorial/part5_gradle_caching.adoc

```diff
@@ -0,0 +1,238 @@
+[source]
+----
+org.gradle.console=verbose
+org.gradle.caching=true
+----
+
+TIP: It's also a good time to add verbose outputs to Gradle by setting `org.gradle.console=verbose`.
+
+== Step 3. Use Local Cache
+Run the clean task and then the build task to populate the cache.
+[source]
+----
+$ ./gradlew :app:clean :app:build
+
+> Task :app:clean
+> Task :app:compileJava
+> Task :app:processResources
+> Task :app:classes
+> Task :app:jar
+> Task :app:startScripts
+> Task :app:distTar
+> Task :app:distZip
+> Task :app:assemble
+> Task :app:compileTestJava
+> Task :app:processTestResources NO-SOURCE
+> Task :app:testClasses
+> Task :app:test
+> Task :app:check
+> Task :app:build
+
+BUILD SUCCESSFUL in 6s
+9 actionable tasks: 9 executed
+----
+
+No surprise here, Gradle invoked all the tasks needed to build the app and it was done successfully.
+
+Let's run the build again.
+[source]
+----
+$ ./gradlew :app:build
+
+> Task :app:compileJava UP-TO-DATE
+> Task :app:processResources UP-TO-DATE
+> Task :app:classes UP-TO-DATE
+> Task :app:jar UP-TO-DATE
+> Task :app:startScripts UP-TO-DATE
+> Task :app:distTar UP-TO-DATE
+> Task :app:distZip UP-TO-DATE
+> Task :app:assemble UP-TO-DATE
+> Task :app:compileTestJava UP-TO-DATE
+> Task :app:processTestResources NO-SOURCE
+> Task :app:testClasses UP-TO-DATE
+> Task :app:test UP-TO-DATE
+> Task :app:check UP-TO-DATE
+> Task :app:build UP-TO-DATE
+----
+
+As we can see, Gradle used the _incremental build_ optimization.
+Nothing has changed so Gradle has decided not to re-run some tasks.
+Instead letting us know that inputs and outputs have not changed using the _outcome label_ `UP-TO-DATE`.
+
+Now let's delete all the files by running the clean and build back to back:
+[source]
+----
+$ ./gradlew :app:clean :app:build
+
+> Task :app:clean
+> Task :app:compileJava FROM-CACHE
+> Task :app:processResources
+> Task :app:classes
+> Task :app:jar
+> Task :app:startScripts
+> Task :app:distTar
+> Task :app:distZip
+> Task :app:assemble
+> Task :app:compileTestJava FROM-CACHE
+> Task :app:processTestResources NO-SOURCE
+> Task :app:testClasses UP-TO-DATE
+> Task :app:test FROM-CACHE
+> Task :app:check UP-TO-DATE
+> Task :app:build
+
+BUILD SUCCESSFUL in 551ms
+9 actionable tasks: 6 executed, 3 from cache
+----
+
+Nothing has changed but files were deleted during the clean task so Gradle has to re-run tasks to re-build the App.
+
+Gradle is efficient, especially with the local cache turn on.
+Gradle will look at the cache directory on your machine to check for output files that may already exist. If they do, instead of running that task, it will copy its (output) results into your project `build` directory.
+
+The _outcome label_ `FROM-CACHE` lets the user know that Gradle has fetched the task results from the local cache.
+
+|===
+|Outcome Label |Description
+
+|UP-TO-DATE
+|Task that has been already executed and hasn't changed (incremental build feature)
+
+|SKIPPED
+|Task has been skipped for some reason
```

Comment:
"for some reason" doesn't sound good.  How about "Task was explicitly prevented from running."